proxy/envoy/config.go

```diff
@@ -503,6 +503,19 @@ func buildInboundListeners(mesh *proxyconfig.ProxyMeshConfig, sidecar proxy.Node
 				Routes:  []*HTTPRoute{route},
 			}
 
+			// Websocket enabled routes need to have an explicit use_websocket : true
+			// This setting needs to be enabled on Envoys at both sender and receiver end
+			if protocol == model.ProtocolHTTP {
+				// get all the route rules applicable to the instances
+				rules := config.RouteRulesByDestination(instances)
```

Comment:
Hm, the target generally doesn't know where the request comes from, so can't truthfully infer the rules that carried the request to destination unless it's in the request metadata.